main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "opensource", "git@github.com:mendersoftware/"+repo+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	gitcmd = exec.Command("git", "remote", "add", "enterprise", "git@github.com:mendersoftware/"+repo+"-enterprise"+".git")
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the branch which we are going to sync
+	gitcmd = exec.Command("git", "fetch", "opensource", branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Fetch the Enterprise branch in which to merge into, and create the PR branch
+	gitcmd = exec.Command("git", "fetch", "enterprise", branchName+":"+PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Checkout the enterprise PR branch
+	gitcmd = exec.Command("git", "checkout", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	// Merge the branches
+	mergeMessage := fmt.Sprintf("Merge branch '%s' into '%s'", "opensource/"+branchName, "enterprise/"+PRBranchName)
+	gitcmd = exec.Command("git", "merge", "-m", mergeMessage, "opensource/"+branchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Debugf("git merge output: %s", out)
+
+	switch string(out) {
+	case "Already up to date.\n":
+		return errors.New("The branches seem to already be in sync, which they are not supposed to be, after a merge to OS!")
+	}
+
+	// Push the branch to enterprise
+	gitcmd = exec.Command("git", "push", "--set-upstream", "enterprise", PRBranchName)
+	gitcmd.Dir = tmpdir
+	out, err = gitcmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+	}
+
+	log.Infof("Merged branch: opensource/%s/%s into enterprise/%[1]s/%s in the Enterprise repo", repo, branchName, PRBranchName)
+
+	return nil
+}
+
+type createPRArgs struct {
+	repo        string
+	prBranch    string
+	baseBranch  string
+	accessToken string
+	message     string
+	messageBody string
+	dryRun      bool
+}
+
+func createPullRequestFromEnterpriseBranch(args createPRArgs) (*github.PullRequest, error) {
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: args.accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
```

Comment:
Can't we resuse `createGitHubClient`?

Reply:
This is related to my very first comment: I think we can reuse the same token, but I might be wrong...

Reply:
If you don't see any issues with reusing it, then I'll join them now.

Reply:
Yes. I think is the best to keep it tidy